lib/generators/controller_override.rb

```diff
@@ -1,17 +1,24 @@
 require "rails/generators"
 require "rails/generators/rails/controller/controller_generator"
 require "rails/generators/rails/scaffold_controller/scaffold_controller_generator"
+require "rails/generators/model_helpers"
 
 module Rails
   module Generators
     class ControllerGenerator
+      include Rails::Generators::ModelHelpers
+
+      class_option :model_name, type: :string, desc: "ModelName to be used"
```

Comment:
Hmm. Basically, `controller` doesn't need to know about `model`. And, the current implementation depends on `controller` name for naming(That's why this file overrides `ControllerGenerator` I think). So overring `controller` name by `model_name` looks strange to me. 
How about using `decorator_name`?

Reply:
`scaffold_controller` already has that interface and I'd prefer `controller` to be uniform with it. Rails already has `--model-name` option and I'd rather use it.

I agree that model dependency for `controller` is dubious, but we already have a `decorator` hook for `controller` that needs to know about a model. That's why `controller` may also need to.

Another option is to remove `decorator` hook from `controller` and add it to `model`. In my opinion, decorators are bound to models rather than controllers or views. However, I'd keep the existing API for now rather than introducing another braking change.

Reply:
I agree that it might be good to hook my models, not controllers. But, it seems that the current behavior is intended #553.
So generating decorators by controller names is the intended behavior(At least, so far). Setting a decorator name by `--model-name` is inconsistent with that behavior I think.

Reply:
> So generating decorators by controller names is the intended behavior.

Yes, it is. And it's not being changed.

Models are also generated based on controller names, but we have `--model-name` to override that when needed.

> `scaffold_controller` already has that interface…

So, why shouldn't a decorator respect it as well when it's specified explicitly?

I'm afraid, I'm missing your point… What is your solution for #919?

Reply:
> Yes, it is. And it's not being changed.

I think this PR changes the behavior. For example, if you run `./bin/rails g scaffold admin/book name:string --model-name book`, the result will be change between `master` branch and this PR.

To be honest, I don't think #919 is a bug. That is the intended behavior of the current generator structure(I can't understand why `draper` only hooks for `controller` though).
But of course, I understand the user's expectations. So adding `decorator_name`(or something like that) to override decorators name is a good solution I think.